File: cpp/141-150/Binary Tree Postorder Traversal.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> result;
        if (root == nullptr) return result;
        stack<TreeNode*> nodes_ready_for_visiting;
        nodes_ready_for_visiting.emplace(root);
        
        while (!nodes_ready_for_visiting.empty())
        {
            root = nodes_ready_for_visiting.top();
            nodes_ready_for_visiting.pop();
            result.emplace(result.end(), root->val);
            
            if (root->left)
                nodes_ready_for_visiting.emplace(root->left);
            if (root->right)
                nodes_ready_for_visiting.emplace(root->right);
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: python/141-150/Binary Tree Postorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {integer[]}
    def postorderTraversal(self, root):
        v = []
        if not root:
            return v
        
        stacks = []
        stacks.append(root)
        
        while len(stacks) > 0:
            root = stacks[-1]
            del stacks[-1]
            v.append(root.val)
            
            if root.left:
                stacks.append(root.left)
            if root.right:
                stacks.append(root.right)
        
        v.reverse()
    return v